components/brave_wallet/browser/json_rpc_service_unittest.cc

```diff
@@ -2480,154 +2532,233 @@ TEST_F(JsonRpcServiceUnitTest, GetERC721Metadata) {
+
+  // ERC1155
+  SetTokenMetadataInterceptor(
+      kERC1155MetadataInterfaceId, mojom::kMainnetChainId,
+      interface_supported_response, https_token_uri_response,
+      https_metadata_response);
+  TestGetTokenMetadata("0x59468516a8259058bad1ca5f8f4bff190d30e066", "0x719",
+                       mojom::kMainnetChainId, kERC1155MetadataInterfaceId,
+                       https_metadata_response, mojom::ProviderError::kSuccess,
+                       "");
+}
+
+TEST_F(JsonRpcServiceUnitTest, GetERC721Metadata) {
+  const std::string https_metadata_response =
+      R"({"attributes":[{"trait_type":"Feet","value":"Green Shoes"},{"trait_type":"Legs","value":"Tan Pants"},{"trait_type":"Suspenders","value":"White Suspenders"},{"trait_type":"Upper Body","value":"Indigo Turtleneck"},{"trait_type":"Sleeves","value":"Long Sleeves"},{"trait_type":"Hat","value":"Yellow / Blue Pointy Beanie"},{"trait_type":"Eyes","value":"White Nerd Glasses"},{"trait_type":"Mouth","value":"Toothpick"},{"trait_type":"Ears","value":"Bing Bong Stick"},{"trait_type":"Right Arm","value":"Swinging"},{"trait_type":"Left Arm","value":"Diamond Hand"},{"trait_type":"Background","value":"Blue"}],"description":"5,000 animated Invisible Friends hiding in the metaverse. A collection by Markus Magnusson & Random Character Collective.","image":"https://rcc.mypinata.cloud/ipfs/QmXmuSenZRnofhGMz2NyT3Yc4Zrty1TypuiBKDcaBsNw9V/1817.gif","name":"Invisible Friends #1817"})";
+
+  // Ensure GetERC721Metadata passes the correct interface ID to
+  // GetTokenMetadata
+  SetTokenMetadataInterceptor(kERC721MetadataInterfaceId,
+                              mojom::kMainnetChainId,
+                              R"({
+                                  "jsonrpc":"2.0",
+                                  "id":1,
+                                  "result": "0x0000000000000000000000000000000000000000000000000000000000000001"
+                              })",
+                              R"({
+                                  "jsonrpc":"2.0",
+                                  "id":1,
+                                  "result":"0x0000000000000000000000000000000000000000000000000000000000000020000000000000000000000000000000000000000000000000000000000000002468747470733a2f2f696e76697369626c65667269656e64732e696f2f6170692f3138313700000000000000000000000000000000000000000000000000000000"
+                              })",
+                              https_metadata_response);
   TestGetERC721Metadata("0x59468516a8259058bad1ca5f8f4bff190d30e066", "0x719",
-                        mojom::kMainnetChainId, "",
-                        mojom::ProviderError::kParsingError,
-                        l10n_util::GetStringUTF8(IDS_WALLET_PARSING_ERROR));
+                        mojom::kMainnetChainId, https_metadata_response,
+                        mojom::ProviderError::kSuccess, "");
+}
+
+TEST_F(JsonRpcServiceUnitTest, GetERC1155Metadata) {
+  const std::string https_metadata_response =
```

Comment:
You can put this either in the anonymous namespace or a member value of this test class and share it.